### ISSUE TITLE
Can't install Horovod Spark

### ISSUE THREAD

**Issue by brandon-biggs** [horovod/horovod#1641]
While trying to follow the installation instructions for Horovod in Spark (https://github.com/horovod/horovod/blob/master/docs/spark.rst) I run into the issue that the Horovod[Spark] packages do not exist. I also noticed that on this documentation page, the required version of Horovod is >= 0.19.0, but the current release of Horovod is 0.18.2. Does Horovod not currently support Spark? I've been looking all over and saw a few other issues related to this, but haven't been able to find any solutions, besides the documentation. Any guidance is appreciated. Thanks.

**Comment by tgaddair** [comment]
Hey @brandon-biggs, please take a look at the Spark docs for [v0.18.2](https://github.com/horovod/horovod/blob/v0.18.2/docs/spark.rst).  The docs you're referring to are from master, which has unreleased changes.  Apologies for the confusion.

**Comment by brandon-biggs** [comment]
I'll give that a shot @tgaddair. Thanks for pointing me to that!

**Comment by brandon-biggs** [comment]
Do I need to be using tensorflow 1.x? The example provided in that documentation uses 'ConfigProto' which apparently has been [deprecated](https://github.com/tensorflow/tensorflow/issues/33504) as of Tensorflow2.x. 

So pointing me to that documentation allowed me to install the packages for Spark, but I wasn't able to get a good test. My main goal here is to run horovod from within a Jupyter Notebook on multiple GPUs. I've gotten it to work from within a bash script using MPI, but haven't gotten it to work yet in a Jupyter notebook. This started with issue #622 and following the trail of trying to use Horovod with Spark. I can comment on that issue if that would be better.

Thanks again.

**Comment by tgaddair** [comment]
Hey @brandon-biggs, you do not need TensorFlow 1.x.  The Spark example provided uses TensorFlow 1, but we also have TensorFlow 2 examples you can take a look at that show how to pin GPUs using their new API (see: [tensorflow2_mnist.py](https://github.com/horovod/horovod/blob/master/examples/tensorflow2_mnist.py)).

If your goal is to run from a Jupyter notebook, and not necessarily to run on top of Spark, then you may want to try installing from the master branch (`pip install --no-cache-dir git+https://github.com/horovod/horovod.git`), and taking a look at [horovod.run.run](https://github.com/horovod/horovod/blob/master/horovod/run/run.py#L870).

**Comment by brandon-biggs** [comment]
@tgaddair horovod.run.run looks like it would be helpful, but I'm not able to install horovod from the master branch. 

Here's some information:
Centos7.7
Conda 4.8
Python 3.7
Tensorflow-GPU 2.0.0

I created the environment with Conda and installed tensorflow with Conda, but am installing horovod with pip inside of the conda environment, following the instructions [here](https://github.com/horovod/horovod/blob/master/docs/gpus.rst). My horovod pip install command is: `HOROVOD_NCCL_HOME=/usr/local/nccl_2.5.6 HOROVOD_GPU_ALLREDUCE=NCCL pip install --no-cache-dir horovod`. That works. But when I try to install the master branch, as you suggested, it fails. I just replace `horovod` with `git+https://github.com/horovod/horovod.git`. I did also try to download the repository locally and tried to checkout a different commit of master that didn't have some failing tests, but that didn't work either. Below are some of my errors.

```
In file included from horovod/common/ops/adasum/adasum_mpi.h:23:0,
                   from horovod/common/ops/adasum/adasum_mpi.cc:16:
  horovod/common/ops/adasum/adasum.h:459:7: note: called from here
         dotProductVec = _mm256_fmadd_pd(aBot, bBot, dotProductVec);
         ^~~~~~~~~~~~~
  INFO: Unable to build TensorFlow plugin, will skip it.
...

error: None of TensorFlow, PyTorch, or MXNet plugins were built.
```

**Comment by tgaddair** [comment]
Hey @brandon-biggs, can you share the full error log in a gist?

**Comment by brandon-biggs** [comment]
@tgaddair Here's a [link](https://gist.github.com/brandon-biggs/08256a8640a51156d7540d9c59500539) to the gist

**Comment by brandon-biggs** [comment]
Hi @tgaddair Do you have any ideas on what the issue might be? We've also tried using ipyparallel to run horovod in a jupyter, but that didn't work out so well.

Multiple nodes with multiple nodes work great with horovod batch. The horovod.run.run seems like it would work, but I was curious if you had any other ideas.

**Comment by tgaddair** [comment]
Hey @brandon-biggs, we released Horovod v0.19.0 yesterday with support for `horovod.run.run`.  Can you give it a shot?

**Comment by brandon-biggs** [comment]
I'm working through this now. Thanks for all the work and releasing this new version. I've ran into some new things, but they're unrelated to the issue I originally created this issue for. I'm going to close this up. I believe that horovod.run.run will solve our problems once I resolve some other, unrelated issues. Thanks again.
